mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/SmartTestingMavenConfigurer.java

```diff
@@ -80,6 +95,6 @@ private void logStrategiesNotDefined() {
     }
 
     private boolean isSkipExtension() {
-        return configuration.isDisabled() || isSkipTestExecutionSet();
+        return configuration.isDisabled() || isSkipTestExecutionSet() || isSpecificTestClassSet();
```

Comment:
How about storing the information that the extension hasn't been loaded? 
The system properties could be tricky as anything could be changed during the test execution. In addition, system properties are not thread-safe, but I'm not sure if we would face this problem in this case.
WDYT @bartoszmajsak

Reply:
I guess you mean logging from the loaded extension that its execution has been skipped. I think simply logging it would make sense. If you agree let's do it like that. Log that we skip the extension and why.

Reply:
@bartoszmajsak, we do this here, https://github.com/arquillian/smart-testing/pull/118/files#diff-d4e45adf3ad3bc619ccc2eef6c35f04bR50.
Do you expect something else ?

Reply:
Ah storing in this sense :) Ok. Sure, it's read only once and never changed during the execution. It's a small optimization though.